src/Kernel32.Tests.Shared/UseCultureAttribute.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using System.Reflection;
+using System.Threading;
+using Xunit.Sdk;
+
+/// <summary>
+/// Apply this attribute to your test method to replace the
+/// Thread.CurrentThread <see cref="CultureInfo.CurrentCulture" /> and
+/// <see cref="CultureInfo.CurrentUICulture" /> with another culture.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+internal class UseCultureAttribute : BeforeAfterTestAttribute
+{
+    private readonly Lazy<CultureInfo> culture;
+    private readonly Lazy<CultureInfo> uiCulture;
+
+#if !PROFILE_111
+    private CultureInfo originalCulture;
+    private CultureInfo originalUICulture;
+#endif
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UseCultureAttribute"/> class.
+    /// It replaces the culture and UI culture of the current thread with
+    /// <paramref name="culture" />
+    /// </summary>
+    /// <param name="culture">The name of the culture.</param>
+    /// <remarks>
+    /// <para>
+    /// This constructor overload uses <paramref name="culture" /> for both
+    /// <see cref="Culture" /> and <see cref="UICulture" />.
+    /// </para>
+    /// </remarks>
+    public UseCultureAttribute(string culture)
+        : this(culture, culture)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UseCultureAttribute"/> class.
+    /// It replaces the culture and UI culture of the current thread with
+    /// <paramref name="culture" /> and <paramref name="uiCulture" />
+    /// </summary>
+    /// <param name="culture">The name of the culture.</param>
+    /// <param name="uiCulture">The name of the UI culture.</param>
+    public UseCultureAttribute(string culture, string uiCulture)
+    {
+        this.culture = new Lazy<CultureInfo>(() => new CultureInfo(culture));
+        this.uiCulture = new Lazy<CultureInfo>(() => new CultureInfo(uiCulture));
+    }
+
+    /// <summary>
+    /// Gets the culture.
```

Comment:
Gets the culture...... that will be set by this attribute.
You know, as opposed to the current culture, or the culture that was current when the test started, etc.